### ISSUE TITLE
Change serde to be optional again

### ISSUE THREAD

**Issue by Pauan** [Pauan/rust-signals#19]

**Comment by mpfaff** [comment]
I think this issue can be closed

**Comment by Pauan** [comment]
serde hasn't been made optional, because that's a breaking change.

**Comment by mpfaff** [comment]
The `Cargo.toml` [appears](https://github.com/Pauan/rust-signals/blob/36585021f8230d74083640609b90480dbf1b6078/Cargo.toml#L31) to say otherwise. Is disabling the `serde` feature by default the breaking change?

**Comment by Pauan** [comment]
Yes, [default features](https://github.com/Pauan/rust-signals/blob/36585021f8230d74083640609b90480dbf1b6078/Cargo.toml#L15) are always applied, so they're effectively required.

Even if you try to use `default-features = false`, [it doesn't really work](https://github.com/rust-lang/cargo/issues/7545#issuecomment-546983311). In order for `default-features = false` to work, *every dependency* has to use it. If even one dependency doesn't use it, then it won't work.

For example, [`dominator` has a dependency on `futures-signals`](https://github.com/Pauan/rust-dominator/blob/623547f4244cf02e1f1016044981cf34649d8127/Cargo.toml#L25). If you use `dominator` and `futures-signals`, then even if you use `default-features = false`, it won't work, because dominator isn't using `default-features = false`.

This means that `default-features = false` is pretty much always useless and does nothing. So in practice default features are always mandatory.
